library/core/src/str/pattern.rs

```diff
@@ -498,6 +514,22 @@ unsafe impl<'a> ReverseSearcher<'a> for CharSearcher<'a> {
     }
     #[inline]
     fn next_match_back(&mut self) -> Option<(usize, usize)> {
+        if self.utf8_size == 1 {
+            let find = |haystack: &[u8]| {
+                if haystack.len() < 64 {
```

Comment:
Is it intended for the if-else conditions there to be inversed compared to `next_match` implementation?

Reply:
oops.